d3a_api_client/rest_device.py

```diff
@@ -0,0 +1,163 @@
+from d3a_api_client import APIClientInterface
+import os
+import logging
+import requests
+import json
+import ssl
+from functools import wraps
+from concurrent.futures.thread import ThreadPoolExecutor
+from d3a_api_client.websocket_device import WebsocketMessageReceiver, WebsocketThread
+
+
+root_logger = logging.getLogger()
+root_logger.setLevel(logging.INFO)
+
+
+def logging_decorator(command_name):
+    def decorator(f):
+        @wraps(f)
+        def wrapped(self, *args, **kwargs):
+            logging.debug(f'Sending command {command_name} to device.')
+            return_value = f(self, *args, **kwargs)
+            logging.debug(f'Command {command_name} responded with: {return_value}.')
+            return return_value
+        return wrapped
+    return decorator
+
+
+class RestDeviceClient(APIClientInterface):
+
+    def __init__(self, simulation_id, device_id, domain_name,
+                 websockets_domain_name, is_ssl, autoregister=False):
+        self.simulation_id = simulation_id
+        self.device_id = device_id
+        self.domain_name = domain_name
+        self.retrieve_key_from_server()
+
+        if is_ssl:
+            self.pem_cert = ssl.get_server_certificate(domain_name)
+        self.dispatcher = WebsocketMessageReceiver(self)
+        self.websocket_thread = WebsocketThread(simulation_id, device_id, self.jwt_token,
+                                                websockets_domain_name, self.dispatcher)
+        self.websocket_thread.start()
+        self.callback_thread = ThreadPoolExecutor(max_workers=5)
+        self.registered = False
+        if autoregister:
+            self.register()
+
+    def retrieve_key_from_server(self):
+        resp = requests.post(
+            f"{self.domain_name}/api-token-auth/",
+            data=json.dumps({"username": os.environ["API_CLIENT_USERNAME"],
+                             "password": os.environ["API_CLIENT_PASSWORD"]}),
+            headers={"Content-Type": "application/json"})
+        if resp.status_code != 200:
+            logging.error(f"Request for token authentication failed with status code {resp.status_code}."
+                          f"Response body: {resp.text}")
+            return
+        self.jwt_token = json.loads(resp.text)["token"]
+
+    @property
+    def _url_prefix(self):
+        return f'{self.domain_name}/external-connection/api/{self.simulation_id}/{self.device_id}/'
+
+    def post_request(self, endpoint_suffix, data):
```

Comment:
I think it could be changed to a private method?

Reply:
Changing it, thanks